docs/GUIDE.md

```diff
@@ -560,24 +589,86 @@ purp = pconstant $ Minting "be"
+This comes in 3 flavors.
+* Plutarch type represented **directly** by a builtin type that **is not** `Data` (`DefaultUniData`) ==> `DerivePConstantViaCoercible`
+
+  Ex: `PInteger` is directly represented as a builtin integer.
+* Plutarch type represented **indirectly** by a builtin type that **is not** `Data` (`DefaultUniData`) ==> `DerivePConstantViaNewtype`
+
+  Ex: `PPubKeyHash` is a newtype to a `PByteString`, `PByteString` is *directly* represented as a builtin bytestring.
+* Plutarch type represented by `Data` (`DefaultUniData`) ==> `DerivePConstantViaData`
+
+  Ex: `PScriptPurpose` is represented as a `Data` value. It is synonymous to `ScriptPurpose` from the Plutus ledger api.
+
+Whichever path you need to go down, there is one common part- implementing `PLift`, or rather `PUnsafeLiftDecl`. See, `PLift` is actually just a type synonym to `PUnsafeLiftDecl`. Essentially an empty typeclass with an associated type family that provides insight on the relationship between a Plutarch type and its Haskell synonym.
+```hs
+instance PUnsafeLiftDecl YourPlutarchType where type PLifted YourPlutarchType = YourHaskellType
+```
+You're tasked with assigning the correct Haskell synonym to your Plutarch type, and what an important task it is! Recall that `pconstant`'s argument type will depend on your assignment here. In particular: `pconstant :: YourHaskellType -> YourPlutarchType`.
+
+Some examples:-
+* for `YourPlutarchType` = `PInteger`, `YourHaskellType` = `Integer`
+
+  ```hs
+  instance PUnsafeLiftDecl PInteger where type PLifted PInteger = Integer
+  ```
+* for `YourPlutarchType` = `PValidatorHash`, `YourHaskellType` = `ValidatorHash`
+
+  ```hs
+  instance PUnsafeLiftDecl PValidatorHash where type PLifted PValidatorHash = Plutus.ValidatorHash
+  ```
+* for `YourPlutarchType` = `PScriptPurpose`, `YourHaskellType` = `ScriptPurpose`
+
+  ```hs
+  instance PUnsafeLiftDecl PScriptPurpose where type PLifted PScriptPurpose = Plutus.ScriptPurpose
+  ```
+
+Now, let's get to implementing `PConstant` for the Haskell synonym, via the 3 methods. The first of which is `DerivePConstantViaCoercible`-
 ```hs
-data PBool s
-  deriving (PLift) via PBuiltinType PBool Bool
+deriving via (DerivePConstantViaCoercible Integer PInteger Integer) instance (PConstant Integer)
 ```
-In this case, `PBool` is represented as a builtin boolean in Plutus Core, as [`Bool`](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Bool.html#t:Bool) corresponds to builtin boolean.
-This lets you do `pconstant True` to obtain a `Term s PBool`. And `plift $ pconstant True` to obtain a `Bool`.
+`DerivePConstantViaCoercible` takes in 3 type parameters-
+* The Haskell type itself, for which `PConstant` is being implemented for.
+* The **direct** Plutarch synonym to the Haskell type.
+* Another Haskell type that the first param can be coerced into. TODO: more info
```

Comment:
@L-as I put a `TODO` here for more information. We should put some more information here before merging. I'm not entirely sure about this 3rd param's significance yet - so feel free to tell me what to put in here.

Reply:
Now that I think about it basically all uses of it would instead use `ViaNewtype`.

Reply:
Alright, updated.

Reply:
> Now that I think about it basically all uses of it would instead use `ViaNewtype`.
Yeah.....